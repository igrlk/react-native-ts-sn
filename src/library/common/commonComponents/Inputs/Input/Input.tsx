import React from 'react';
import { TextInput, Text, View } from 'react-native';

import styles from './inputStyles';

interface InputProps {
	label?: string;
	value: string;
	onChange: (value: string) => void;
	style?: { [key: string]: any };
	rightLabel?: any;
	error?: string | null;
	multiline?: boolean;
	numberOfLines?: number;
	theme?: 'common' | 'dark';
}

export default function Input({
	label,
	value,
	onChange,
	style: extraStyle,
	rightLabel,
	error,
	multiline,
	numberOfLines,
	theme,
}: InputProps) {
	return (
		<View style={extraStyle}>
			<View style={styles.labelContainer}>
				{label && <Text style={[styles.label, theme === 'dark' && styles.labelDark]}>{label}</Text>}
				{rightLabel}
			</View>
			<TextInput
				autoCapitalize='none'
				value={value}
				onChangeText={onChange}
				style={[styles.input, multiline && styles.textArea, theme === 'dark' && styles.inputDark]}
				multiline={multiline}
				numberOfLines={numberOfLines}
			/>
			{error && <Text style={styles.error}>{error}</Text>}
		</View>
	);
}

Input.defaultProps = {
	style: {},
};
