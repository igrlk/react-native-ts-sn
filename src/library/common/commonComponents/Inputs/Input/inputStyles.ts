import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	input: {
		height: 40,
		width: '100%',
		paddingLeft: 12,
		borderRadius: 2,
		borderWidth: 1,
		borderColor: '#EBEBEB',
		backgroundColor: '$colorWhite',
		fontSize: 12,
		color: '$colorBlack',
		display: 'flex',
		alignItems: 'flex-start',
		justifyContent: 'center',
		zIndex: 0,
	},
	inputDark: {
		height: 40,
		borderRadius: 2,
		borderColor: '#9E9E9E',
		paddingLeft: 10,
		paddingRight: 10,
		color: '$colorBlack',
		fontFamily: 'Poppins',
		fontSize: 16,
		lineHeight: 24,
		borderWidth: 1,
	},
	label: {
		fontFamily: 'Poppins',
		fontSize: 12,
		lineHeight: 18,
	},
	labelDark: {
		marginBottom: -3,
		color: '#9E9E9E',
		fontFamily: 'Poppins',
		fontSize: 14,
		lineHeight: 21,
	},
	labelContainer: {
		display: 'flex',
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'flex-end',
	},
	error: {
		fontFamily: 'Poppins',
		fontSize: 10,
		lineHeight: 15,
		color: '$colorRed',
		marginTop: 2,
	},
	textArea: {
		height: 86,
	},
});
