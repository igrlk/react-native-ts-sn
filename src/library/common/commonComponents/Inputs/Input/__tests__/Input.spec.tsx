import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Input from '../Input';

describe('Input', () => {
	let props = {
		label: 'some-label',
		value: 'some-value',
		onChange: jest.fn(),
		rightLabel: 'some-right-label',
		error: 'some-error',
		multiline: true,
		numberOfLines: 4,
	};
	beforeEach(
		() =>
			(props = {
				label: 'some-label',
				value: 'some-value',
				onChange: jest.fn(),
				rightLabel: 'some-right-label',
				error: 'some-error',
				multiline: true,
				numberOfLines: 4,
			}),
	);
	it('match to snapshot', () => {
		const tree = renderer.create(<Input {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should call navigator.goBack by clicking on touchable opacity', () => {
		const onClick = jest.fn();
		const component = shallow(<Input {...props} />);
		component.find('TextInput').simulate('changeText');
		expect(props.onChange).toHaveBeenCalled();
	});
});
