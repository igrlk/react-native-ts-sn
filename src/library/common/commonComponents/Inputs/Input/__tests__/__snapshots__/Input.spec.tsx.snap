// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input match to snapshot 1`] = `
<View
  style={Object {}}
>
  <View>
    <Text
      style={
        Array [
          undefined,
          false,
        ]
      }
    >
      some-label
    </Text>
    some-right-label
  </View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    multiline={true}
    numberOfLines={4}
    onChangeText={[MockFunction]}
    rejectResponderTermination={true}
    style={
      Array [
        undefined,
        undefined,
        false,
      ]
    }
    underlineColorAndroid="transparent"
    value="some-value"
  />
  <Text>
    some-error
  </Text>
</View>
`;
